src/main/scala/it/pagopa/interop/agreementprocess/common/Adapters.scala

```diff
@@ -10,24 +10,79 @@ import java.util.UUID
 
 object Adapters {
 
-  val ACTIVABLE_STATES: Set[PersistentAgreementState]   =
+  val ACTIVABLE_STATES: Set[AgreementManagement.AgreementState]   =
+    Set(AgreementManagement.AgreementState.PENDING, AgreementManagement.AgreementState.SUSPENDED)
+  val SUSPENDABLE_STATES: Set[AgreementManagement.AgreementState] =
+    Set(AgreementManagement.AgreementState.ACTIVE, AgreementManagement.AgreementState.SUSPENDED)
+  val ARCHIVABLE_STATES: Set[AgreementManagement.AgreementState]  =
+    Set(AgreementManagement.AgreementState.ACTIVE, AgreementManagement.AgreementState.SUSPENDED)
+  val SUBMITTABLE_STATES: Set[AgreementManagement.AgreementState] = Set(AgreementManagement.AgreementState.DRAFT)
+  val UPDATEABLE_STATES: Set[AgreementManagement.AgreementState]  =
+    Set(AgreementManagement.AgreementState.DRAFT)
+  val UPGRADABLE_STATES: Set[AgreementManagement.AgreementState]  =
+    Set(AgreementManagement.AgreementState.ACTIVE, AgreementManagement.AgreementState.SUSPENDED)
+  val REJECTABLE_STATES: Set[AgreementManagement.AgreementState]  = Set(AgreementManagement.AgreementState.PENDING)
+  val DELETABLE_STATES: Set[AgreementManagement.AgreementState]   =
+    Set(AgreementManagement.AgreementState.DRAFT, AgreementManagement.AgreementState.MISSING_CERTIFIED_ATTRIBUTES)
+
+  val PERSISTENT_ACTIVABLE_STATES: Set[PersistentAgreementState]   =
     Set(Pending, Suspended)
-  val SUSPENDABLE_STATES: Set[PersistentAgreementState] =
+  val PERSISTENT_SUSPENDABLE_STATES: Set[PersistentAgreementState] =
     Set(Active, Suspended)
-  val ARCHIVABLE_STATES: Set[PersistentAgreementState]  =
+  val PERSISTENT_ARCHIVABLE_STATES: Set[PersistentAgreementState]  =
     Set(Active, Suspended)
-  val SUBMITTABLE_STATES: Set[PersistentAgreementState] = Set(Draft)
-  val UPDATEABLE_STATES: Set[PersistentAgreementState]  =
+  val PERSISTENT_SUBMITTABLE_STATES: Set[PersistentAgreementState] = Set(Draft)
+  val PERSISTENT_UPDATEABLE_STATES: Set[PersistentAgreementState]  =
     Set(Draft)
-  val UPGRADABLE_STATES: Set[PersistentAgreementState]  =
+  val PERSISTENT_UPGRADABLE_STATES: Set[PersistentAgreementState]  =
     Set(Active, Suspended)
-  val REJECTABLE_STATES: Set[PersistentAgreementState]  = Set(Pending)
-  val DELETABLE_STATES: Set[PersistentAgreementState]   =
+  val PERSISTENT_REJECTABLE_STATES: Set[PersistentAgreementState]  = Set(Pending)
+  val PERSISTENT_DELETABLE_STATES: Set[PersistentAgreementState]   =
     Set(Draft, MissingCertifiedAttributes)
```

Comment:
the duplication of these states can be problematic during the maintenance.
If one of the two sets is no more used, feel free to remove it (I suggest to leave the old names though)